Core/Src/can_handler.c

```diff
@@ -1,11 +1,69 @@
-#include "analyzer.h"
-#include "ringbuffer.h"
 #include "can_handler.h"
-#include "shep_tasks.h"
 #include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
 
-ringbuffer_t *can1_rx_queue = NULL;
-ringbuffer_t *can2_rx_queue = NULL;
+#define CAN_MSG_QUEUE_SIZE 50 /* messages */
+
+#define CAN_DISPATCH_FLAG 1U
+
+#define NEW_CAN_MSG_FLAG 1U
+
+static osMessageQueueId_t can_outbound_queue;
+static osMessageQueueId_t can_inbound_queue;
+
+can_t *can1;
+can_t *can2;
+
+static uint32_t can1_id_list[] = {
+	//CANID_X,
+	0x002
+};
+
+static uint32_t can2_id_list[] = {
+	//CANID_X,
+	0x18FF50E5
+};
+
+osStatus_t queue_and_set_flag(osMessageQueueId_t queue, const void *msg_ptr,
+			      osThreadId_t thread_id, uint32_t flags)
+{
+	osStatus_t status = osMessageQueuePut(queue, msg_ptr, 0U, 0U);
+	osThreadFlagsSet(thread_id, flags);
+	return status;
+}
+
+void init_both_can(CAN_HandleTypeDef *hcan1, CAN_HandleTypeDef *hcan2)
+{
+	assert(hcan1);
+	assert(hcan2);
+
+	can1 = malloc(sizeof(can_t));
+	assert(can1);
+	can2 = malloc(sizeof(can_t));
+	assert(can2);
+
+	can1->hcan = hcan1;
+
+	uint32_t can1_id_list_size_four[4] = { can1_id_list[0], can1_id_list[0],
```

Comment:
Why is this array repeating the same element 4 times?

Reply:
The stm docs require 4 can IDs per filter.  We dont really know how or what will happen if we leave one blank, so we duplicate them.